src/nft/components/profile/list/SetDurationModal.tsx

```diff
@@ -0,0 +1,141 @@
+import ms from 'ms.macro'
+import { SortDropdown } from 'nft/components/common/SortDropdown'
+import { Column, Row } from 'nft/components/Flex'
+import { NumericInput } from 'nft/components/layout/Input'
+import { bodySmall, buttonTextMedium, caption } from 'nft/css/common.css'
+import { useSellAsset } from 'nft/hooks'
+import { DropDownOption } from 'nft/types'
+import { pluralize } from 'nft/utils/roundAndPluralize'
+import { useEffect, useMemo, useState } from 'react'
+import { AlertTriangle } from 'react-feather'
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+const ModalWrapper = styled(Column)`
+  gap: 4px;
+`
+
+const InputWrapper = styled(Row)<{ isInvalid: boolean }>`
+  padding: 12px 8px 12px 12px;
+  border: 1px solid;
+  position: relative;
+  height: 44px;
+  border-radius: 8px;
+  border-color: ${({ isInvalid, theme }) => (isInvalid ? theme.accentCritical : theme.backgroundOutline)};
+`
+
+const DropdownWrapper = styled(ThemedText.BodyPrimary)`
+  cursor: pointer;
+  display: flex;
+  justify-content: flex-end;
+  padding-top: 24px;
+  width: 80px;
+`
+
+const ErrorMessage = styled(Row)`
+  color: ${({ theme }) => theme.accentCritical};
+  gap: 4px;
+`
+
+const WarningIcon = styled(AlertTriangle)`
+  width: 16px;
+  color: ${({ theme }) => theme.accentCritical};
+`
+
+enum Duration {
+  hour = 'hour',
+  day = 'day',
+  week = 'week',
+  month = 'month',
+}
+
+enum ErrorState {
+  valid,
+  empty,
+  overMax,
+}
+
+export const SetDurationModal = () => {
+  const [duration, setDuration] = useState(Duration.day)
+  const [displayDuration, setDisplayDuration] = useState(Duration.day)
+  const [amount, setAmount] = useState(7)
+  const [errorState, setErrorState] = useState(ErrorState.valid)
+  const setGlobalExpiration = useSellAsset((state) => state.setGlobalExpiration)
+  const setCustomExpiration = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAmount(!!event.target.value.length ? parseFloat(event.target.value) : 0)
+    setDuration(displayDuration)
+  }
+  const selectDuration = (duration: Duration) => {
+    setDuration(duration)
+    setDisplayDuration(duration)
+  }
+  const durationOptions: DropDownOption[] = useMemo(
+    () => [
+      {
+        displayText: 'Hours',
+        onClick: () => selectDuration(Duration.hour),
+      },
+      {
+        displayText: 'Days',
+        onClick: () => selectDuration(Duration.day),
+      },
+      {
+        displayText: 'Weeks',
+        onClick: () => selectDuration(Duration.week),
+      },
+      {
+        displayText: 'Months',
+        onClick: () => selectDuration(Duration.month),
+      },
+    ],
+    []
+  )
+  useEffect(() => {
+    const expiration = convertDurationToExpiration(amount, duration)
+    if (expiration * 1000 - Date.now() < ms`60 seconds`) setErrorState(ErrorState.empty)
```

Comment:
Question: do we maintain a general guideline as far as if statements? Shall we skip `{}`? If yes, when?

Reply:
No official guidelines, I generally will skip for 1 line, simple executions. If readability is compromised I'll add braces